engine/state_stream/api_test.go

```diff
@@ -0,0 +1,120 @@
+	"github.com/stretchr/testify/suite"
+
+	"github.com/onflow/flow-go/engine/common/rpc/convert"
+	"github.com/onflow/flow-go/ledger"
+	"github.com/onflow/flow-go/ledger/common/testutils"
+	"github.com/onflow/flow-go/module/executiondatasync/execution_data"
+	downloadermock "github.com/onflow/flow-go/module/executiondatasync/execution_data/mock"
+	protocol "github.com/onflow/flow-go/state/protocol/mock"
+	"github.com/onflow/flow-go/storage"
+	storagemock "github.com/onflow/flow-go/storage/mock"
+	"github.com/onflow/flow-go/utils/unittest"
+)
+
+type Suite struct {
+	suite.Suite
+
+	blocks     *storagemock.Blocks
+	headers    *storagemock.Headers
+	seals      *storagemock.Seals
+	results    *storagemock.ExecutionResults
+	downloader *downloadermock.Downloader
+}
+
+func TestHandler(t *testing.T) {
+	suite.Run(t, new(Suite))
+}
+
+func (suite *Suite) SetupTest() {
+	rand.Seed(time.Now().UnixNano())
+	header := unittest.BlockHeaderFixture()
+	params := new(protocol.Params)
+	params.On("Root").Return(header, nil)
+	suite.blocks = new(storagemock.Blocks)
+	suite.headers = new(storagemock.Headers)
+	suite.seals = new(storagemock.Seals)
+	suite.results = new(storagemock.ExecutionResults)
+
+	suite.downloader = new(downloadermock.Downloader)
+}
+
+func (suite *Suite) TestGetExecutionDataByBlockID() {
+
+	// create the handler with the mock
+	client := New(suite.headers, suite.seals, suite.results, suite.downloader)
+
+	// mock parameters
+	ctx := context.Background()
+	blockHeader := unittest.BlockHeaderFixture()
+
+	seals := unittest.BlockSealsFixture(1)[0]
+	results := unittest.ExecutionResultFixture()
+	numChunks := 5
+	minSerializedSize := 5 * execution_data.DefaultMaxBlobSize
+	chunks := make([]*execution_data.ChunkExecutionData, numChunks)
+
+	for i := 0; i < numChunks; i++ {
+		chunks[i] = generateChunkExecutionData(suite.T(), uint64(minSerializedSize))
+	}
+
+	execData := &execution_data.BlockExecutionData{
+		BlockID:             blockHeader.ID(),
+		ChunkExecutionDatas: chunks,
+	}
+
+	execDataRes, err := convert.BlockExecutionDataToMessage(execData)
+	require.Nil(suite.T(), err)
+
+	suite.headers.On("ByBlockID", blockHeader.ID()).Return(blockHeader, nil)
+	suite.seals.On("FinalizedSealForBlock", blockHeader.ID()).Return(seals, nil)
+	suite.results.On("ByID", seals.ResultID).Return(results, nil)
+	suite.Run("happy path TestGetExecutionDataByBlockID success", func() {
+		suite.downloader.On("Download", ctx, results.ExecutionDataID).Return(execData, nil).Once()
+		res, err := client.GetExecutionDataByBlockID(ctx, blockHeader.ID())
+		suite.downloader.AssertExpectations(suite.T())
+		assert.Equal(suite.T(), execDataRes, res)
+		assert.NoError(suite.T(), err)
+	})
+
+	suite.Run("missing exec data for TestGetExecutionDataByBlockID failure", func() {
+		suite.downloader.On("Download", ctx, results.ExecutionDataID).Return(nil, storage.ErrNotFound).Once()
+		_, err := client.GetExecutionDataByBlockID(ctx, blockHeader.ID())
+		suite.Require().Error(err)
```

Comment:
please assert the specific error as well as asserting `nil` or the empty data you're expecting here